CMakeLists.txt

```diff
@@ -0,0 +1,32 @@
+cmake_minimum_required(VERSION 3.10)
+
+project(UriParser VERSION 0.9.1 LANGUAGES C)
+
+file(GLOB INTERFACE_FILES
+        RELATIVE "${CMAKE_CURRENT_SOURCE_DIR}" CONFIGURE_DEPENDS
+        "include/uriparser/*.h")
+
+file(GLOB SOURCE_FILES
+        RELATIVE "${CMAKE_CURRENT_SOURCE_DIR}" CONFIGURE_DEPENDS
+        "src/*.[c|h]")
```

Comment:
FYI I found this warning at https://cmake.org/cmake/help/v3.0/command/file.html:
> (We do not recommend using GLOB to collect a list of source files from your source tree. If no CMakeLists.txt file changes when a source is added or removed then the generated build system cannot know when to ask CMake to regenerate.)

Reply:
This warning is valid unless you use the new "CONFIGURE_DEPENDS" option which supersedes it.
However, I do in fact tend to name each source file when building larger projects. These days it's become a matter of taste.
Globs can make the build system DRYer but do run the risk of including a file you didn't intend to be there.

Reply:
Interesting!
It turns out that `CONFIGURE_DEPENDS` does not exist prior to 3.12 which is more recent than Ubuntu xenial, which would be at least some problem, e.g. regarding CI. I'll keep it in mind, Let's go with a whitelist for now.